go/oasis-test-runner/cmd/root.go

```diff
@@ -73,17 +99,194 @@ func Execute() {
 }
 
 // RegisterNondefault adds a scenario to the runner.
-func RegisterNondefault(scenario scenario.Scenario) error {
-	n := strings.ToLower(scenario.Name())
-	if _, ok := scenarioMap[n]; ok {
-		return errors.New("root: scenario already registered: " + n)
+func RegisterNondefault(s scenario.Scenario) error {
+	n := strings.ToLower(s.Name())
+	if _, ok := Scenarios[n]; ok {
+		return fmt.Errorf("root: scenario already registered: %s", n)
+	}
+
+	Scenarios[n] = s
+
+	params := s.Parameters()
+	if len(params) > 0 {
+		for k, v := range scenario.ParametersToStringMap(params) {
+			// Populate TestParamsFlags with test parameters and (re-)register it.
+			param := fmt.Sprintf(TestParamsMask, n, k)
+			TestParamsFlags.StringSlice(param, []string{v}, fmt.Sprintf("value(s) of parameter %s for test %s", k, n))
```

Comment:
Instead of manually parsing all parameters from/to strings the flags could be defined in scenarios and only bound here.

Reply:
More concretely: the `Parameters()` could be a map of that would contain a struct for each parameter which could include additional info (such as flag definition)